File: VB/DXSample/ViewModel.vb
Imports System.Collections.ObjectModel

Namespace DXSample

    Public Class MainViewModel

        Public Property Pages As ObservableCollection(Of PageViewModel)

        Public Sub New()
            Pages = New ObservableCollection(Of PageViewModel)() From {New PageViewModel() With {.Text = "Welcome Page", .Type = PageType.Welcome}, New PageViewModel() With {.Text = "Common Page", .Type = PageType.Common}, New PageViewModel() With {.Text = "Completion Page", .Type = PageType.Completion}}
        End Sub
    End Class

    Public Enum PageType
        Welcome
        Common
        Completion
    End Enum

    Public Class PageViewModel

        Public Property Text As String

        Public Property Type As PageType
    End Class
End Namespace

File: CS/DXSample/ViewModel.cs
using DevExpress.Mvvm;
using System;
using System.Collections.ObjectModel;
using System.Linq;

namespace DXSample {
    public class MainViewModel {
        public ObservableCollection<PageViewModel> Pages { get; set; }
        public MainViewModel() {
            Pages = new ObservableCollection<PageViewModel>() {
                new PageViewModel() { Text = "Welcome Page", Type = PageType.Welcome },
                new PageViewModel() { Text = "Common Page", Type = PageType.Common },
                new PageViewModel() { Text = "Completion Page", Type = PageType.Completion },
            };
        }
    }

    public enum PageType { Welcome, Common, Completion }
    public class PageViewModel {
        public string Text { get; set; }    
        public PageType Type { get; set; }        
    }
}
